src/vm/gccover.cpp

```diff
@@ -1745,23 +1816,34 @@ void DoGcStress (PCONTEXT regs, MethodDesc *pMD)
 
     CONSISTENCY_CHECK(!pThread->HasPendingGCStressInstructionUpdate());
 
-#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_) || defined(_TARGET_ARM_) || defined(_TARGET_ARM64_)
-    if (afterCallProtect) 
+    if (numberOfRegs != 0)
     {
-#ifdef _TARGET_AMD64_
-        regs->Rax = retVal;
+        if (afterCallProtect[0])
+        {
+#if defined(_TARGET_AMD64_)
+            regs->Rax = *(DWORD64*)&retValRegs[0]; // I should cut my own hands for writing this.
```

Comment:
bit_cast<> from c+20 could help here, but I fell like I am missing a nice and simple solution here.

Reply:
Use `OBJECTREFToObject` and cast the resulting `Object*` to `DWORD64`?

Reply:
Do you mean `regs->Rax = (DWORD64)OBJECTREFToObject(retValRegs[0]);`